src/main/java/com/minecolonies/coremod/colony/requestsystem/data/StandardRequestableTypeRequestResolverAssignmentDataStore.java

```diff
@@ -0,0 +1,127 @@
+        {
+            return new StandardRequestableTypeRequestResolverAssignmentDataStore();
+        }
+
+        @NotNull
+        @Override
+        public NBTTagCompound serialize(
+          @NotNull final IFactoryController controller,
+          @NotNull final StandardRequestableTypeRequestResolverAssignmentDataStore standardRequestableTypeRequestResolverAssignmentDataStore)
+        {
+            NBTTagCompound compound = new NBTTagCompound();
+
+            compound.setTag(NbtTagConstants.TAG_TOKEN, controller.serialize(standardRequestableTypeRequestResolverAssignmentDataStore.id));
+            compound.setTag(NbtTagConstants.TAG_LIST, standardRequestableTypeRequestResolverAssignmentDataStore.assignments.keySet().stream().map(t -> {
+                NBTTagCompound entryCompound = new NBTTagCompound();
+
+                entryCompound.setTag(NbtTagConstants.TAG_TOKEN, controller.serialize(t));
+                entryCompound.setTag(NbtTagConstants.TAG_LIST, standardRequestableTypeRequestResolverAssignmentDataStore.assignments.get(t).stream()
+                                                                 .map(StandardFactoryController.getInstance()::serialize)
+                                                                 .collect(NBTUtils.toNBTTagList()));
+
+                return entryCompound;
+            }).collect(NBTUtils.toNBTTagList()));
+
+            return compound;
+        }
+
+        @NotNull
+        @Override
+        public StandardRequestableTypeRequestResolverAssignmentDataStore deserialize(@NotNull final IFactoryController controller, @NotNull final NBTTagCompound nbt)
+          throws Throwable
+        {
+            IToken<?> token = controller.deserialize(nbt.getCompoundTag(NbtTagConstants.TAG_TOKEN));
+            Map<TypeToken<?>, Collection<IToken<?>>> map = NBTUtils.streamCompound(nbt.getTagList(NbtTagConstants.TAG_LIST, Constants.NBT.TAG_COMPOUND))
+                                                          .map(nbtTagCompound -> {
+                                                              final TypeToken<?> elementToken = controller.deserialize(nbtTagCompound.getCompoundTag(NbtTagConstants.TAG_TOKEN));
+                                                              final Collection<IToken<?>> elements = NBTUtils.streamCompound(nbtTagCompound.getTagList(NbtTagConstants.TAG_LIST,
+                                                                Constants.NBT.TAG_COMPOUND)).map(elementCompound -> (IToken<?>) controller.deserialize(elementCompound))
+                                                                                                       .collect(Collectors.toList());
+
+                                                              return new Tuple<>(elementToken, elements);
+                                                          }).collect(Collectors.toMap(t -> t.getFirst(), t -> t.getSecond()));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1612)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1612)